### ISSUE TITLE
Changes in the toolbox with lasted caffe

### ISSUE THREAD

**Issue by x1155665** [yosinski/deep-visualization-toolbox#140]
@arikpoz Thank you very much for making the toolbox functional with the new version of caffe :+1: 

I've seen that you have made many changes in terms of usages, which you didn't mentioned in the [README.md](https://github.com/arikpoz/deep-visualization-toolbox/blob/master/README.md).
The two main changes I‘ve found:
1. The local settings are removed to the folder 'model_settings"
2. The locations of the pre-computed jpgs are moved to the caffevis_outputs_dir. And the storage structure is also changed. This results in a necessary preprocess of the jpgs that fetch.sh downloads. 

Are there any other changes that are worth noticing in terms of using the toolbox?

I hope you can mention the usage changes in the README, and provide a solution for the second change I mentioned above, so that the toolbox can be fully functional out-of-box.

**Comment by arikpoz** [comment]
There are plenty of new visualization and many other improvements to the tool, I haven't had the time to update the README, but you can see a "short" summary in the pull request I've created for pushing it all to the original branch: https://github.com/yosinski/deep-visualization-toolbox/pull/128

Regarding the tool to relocate the jpgs, I'm not sure I follow. My version of the tool knows how to load both the previous type of outputs and the updated outputs. Admittedly, I haven't tested the old outputs for a while and it may require an update. In any case, your fix should probably only change the parts around **caffevis_outputs_dir_folder_format == 'original_combined_single_image'**, and not **caffevis_outputs_dir_folder_format == 'max_tracker_output'** which is the new format.

**Comment by x1155665** [comment]
Thanks for pointing me to the summary.

Sorry, I didn't know the existence of **caffevis_outputs_dir_folder_format** when opening this issue. But since 

> patterns view with maximal input images is not implemented when settings.caffevis_outputs_dir_folder_format == 'original_combined_single_image'"

, my fix can be a workaround for the user to load the caffenet-yos without having much trouble.

**Comment by arikpoz** [comment]
I've updated the README file with some examples on what's new

**Comment by arikpoz** [comment]
Regarding your fix, it seems it will break the loading of new outputs

**Comment by x1155665** [comment]
Thanks for the updates!

Sorry, I did not concerned about the new outputs. my intent was simply to load the images that [fetch.sh](https://github.com/arikpoz/deep-visualization-toolbox/blob/master/models/caffenet-yos/fetch.sh) downloads.
How does my fix break the loading of new output? By changing 'maxim*.png' to 'maxim*.jpg' and 'deconv*.png' to 'deconv*.jpg'? I did that just because the format of the downloaded caffenet-yos images is jpg. I can write a new fix later, in which the format will be changed back to png.

**Comment by x1155665** [comment]
The fixed fix is now in the pull request

**Comment by ZhuweiQin** [comment]
Hi,   @x1155665 I encountered some troubles when I try to show the per unit optimize visualization by using the new toolbox. For example, I want to see some unit in the conv1 of CaffeNet. The terminal says "loading opt*.jpg images for layer conv1 channel 90 out of 96", and I can see nothing on the top right corner of this toolbox. I wonder how to solve this problem, and any suggestions will be greatly appreciated. By the way, can I use this toolbox to see the per unit optimize visualization of the ResNet?

**Comment by x1155665** [comment]
Hi, @ZhuweiQin The per unit optimize visualization of CaffeNet-yos is downloaded by [fetch.sh](https://github.com/arikpoz/deep-visualization-toolbox/blob/master/models/caffenet-yos/fetch.sh) . Since this version of DeepVis toolbox has a lot of changes, you may need to relocate the jpgs so that the toolbox can find them. This [tool](https://github.com/arikpoz/deep-visualization-toolbox/pull/2/commits/d4389622e7a3b0cd22fafabdf79088e99b005514) can help you do that. 
You can use it on ResNet, [this ](https://github.com/yosinski/deep-visualization-toolbox/blob/master/doc/computing_per_unit_visualizations.md)will help you.

**Comment by ZhuweiQin** [comment]
Thanks for your kind reminder. @x1155665 So, you mean, I should synthesize the per unit optimize visualization of ResNet by using this [tool](https://github.com/yosinski/deep-visualization-toolbox/blob/master/doc/computing_per_unit_visualizations.md) and then change the [code](https://github.com/arikpoz/deep-visualization-toolbox/pull/2/commits/d4389622e7a3b0cd22fafabdf79088e99b005514) to load the pre-synthesized jpgs, Right?

**Comment by x1155665** [comment]
@ZhuweiQin  Yes and no. The produced images will be placed in the right order, you don't need to relocate them. The relocating tool is just for the downloaded images (CaffeNet-yos).

**Comment by ZhuweiQin** [comment]
@x1155665 Thanks a lot. Let me try.
